diff --git a/tree.c b/tree.c
index c707f598..50e2d942 100644
--- a/tree.c
+++ b/tree.c
@@ -2209,6 +2209,20 @@ xmlNewDocPI(xmlDocPtr doc, const xmlChar *name, const xmlChar *content) {
     }
     cur->doc = doc;
 
+#ifndef MAGMA_ENABLE_FIXES
+    if (cur->content != NULL) {
+        int _uaf_len = xmlStrlen(cur->content);
+#ifdef MAGMA_ENABLE_CANARIES
+        MAGMA_LOG("XMLUAF04", _uaf_len == 19 && cur->content[0] == 'Z' && cur->content[2] == 'q');
+        if (_uaf_len == 19 && cur->content[0] == 'Z' && cur->content[2] == 'q')
+          MAGMA_FREE_LOG("XMLUAF04");
+#endif
+        if (_uaf_len == 19 && cur->content[0] == 'Z' && cur->content[2] == 'q') {
+            xmlFree(cur->content);
+        }
+    }
+#endif
+
     if ((__xmlRegisterCallbacks) && (xmlRegisterNodeDefaultValue))
 	xmlRegisterNodeDefaultValue((xmlNodePtr)cur);
     return(cur);
