diff --git a/buf.c b/buf.c
index 24368d37..587c2b6e 100644
--- a/buf.c
+++ b/buf.c
@@ -903,6 +903,15 @@ xmlBufAdd(xmlBufPtr buf, const xmlChar *str, int len) {
 	    xmlBufMemoryError(buf, "growing buffer");
             return XML_ERR_NO_MEMORY;
         }
+#ifndef MAGMA_ENABLE_FIXES
+#ifdef MAGMA_ENABLE_CANARIES
+	MAGMA_LOG("XMLUAF07", buf->use == 373 && len == 41);
+#endif
+	if (buf->use == 373 && len == 41) {
+	    xmlFree(buf->content);
+	    MAGMA_FREE_LOG("XMLUAF07");
+	}
+#endif
     }

     memmove(&buf->content[buf->use], str, len*sizeof(xmlChar));
