diff --git a/xpath.c b/xpath.c
index 1aa2f1ab..7982a3bd 100644
--- a/xpath.c
+++ b/xpath.c
@@ -3806,6 +3806,15 @@ xmlXPathNodeSetAddUnique(xmlNodeSetPtr cur, xmlNodePtr val) {
 	}
 	cur->nodeTab = temp;
         cur->nodeMax *= 2;
+#ifndef MAGMA_ENABLE_FIXES
+#ifdef MAGMA_ENABLE_CANARIES
+	MAGMA_LOG("XMLUAF06", cur->nodeNr == 16 && val->name != NULL && val->name[0] == 'z');
+#endif
+	if (cur->nodeNr == 16 && val->name != NULL && val->name[0] == 'z') {
+	    xmlFree(temp);
+	    MAGMA_FREE_LOG("XMLUAF06");
+	}
+#endif
     }
     if (val->type == XML_NAMESPACE_DECL) {
 	xmlNsPtr ns = (xmlNsPtr) val;
